tools/Tasks.ps1

```diff
@@ -0,0 +1,200 @@
+﻿$taskScriptDir = [System.IO.Path]::GetDirectoryName($PSCommandPath)
+$env:repoRoot = [System.IO.Path]::GetDirectoryName($taskScriptDir)
+
+#$userPreferenceDir = $env:psuserpreferences
+$userPsFileEnvVariable = $env:psuserpreferences
+$userPsFileDir = "$env:USERPROFILE\psFiles"
+
+[string]$envVariableName="TEST_CSM_ORGID_AUTHENTICATION"
+
+<#
+We allow users to include any helper powershell scripts they would like to include in the current session
+Currently we support two ways to include helper powershell scripts
+1) psuserspreferences environment variable
+2) $env:USERPROFILE\psFiles directory
+We will include all *.ps1 files from any of the above mentioned locations
+#>
+if([System.IO.Directory]::Exists($userPsFileEnvVariable))
+{
+    Get-ChildItem $userPsFileEnvVariable | WHERE {$_.Name -like "*.ps1"} | ForEach {
+    Write-Host "Including $_" -ForegroundColor Green
+    . $userPsFileEnvVariable\$_
+    }
+}
+elseif([System.IO.Directory]::Exists($userPsFileDir))
+{
+    Get-ChildItem $userPsFileDir | WHERE {$_.Name -like "*.ps1"} | ForEach {
+    Write-Host "Including $_" -ForegroundColor Green
+    . $userPsFileDir\$_
+    }
+}
+else
+{
+    Write-Host "Loading skipped. 'psuserpreferences' environment variable was not set to load user preferences." -ForegroundColor DarkYellow
+}
+
+Function Get-TaskHelp
+{
+    $taskCount = 1
+    #Add tasks and description in the tasks array
+    [string[]] $tasks = "Setup-TestEnvironment ==> will allow you to setup Test Environment", `
+                        "List-BuildScopes ==> List of available Scopes", `
+                        "Build-Repo ==> will allow you to build the entire repo or build for specific Scope", `                        
+                        "Run-CheckinTests ==> Run Tests prior to checkin", `
+                        "Run-MockedScenarioTests ==> Run Mocked Scenario Tests", `
+                        "Get-TaskHelp ==> Display supported tasks"
+    
+    #$tableFormat = @{Expression={$taskCount};width=50}, @{Expression={$_}};
+    #$tasks | Format-Table $tableFormat
+    $tasks |foreach { Write-Host ([string]::Format("{0}) {1}", $taskCount++, $_))}
+    Write-Host ""
+}
+
+Function Setup-TestEnvironment()
```

Comment:
Converted to a powershell module and got rid of all of the Read-Host calls for prompts.